### PR TITLE
[megatron] fix: add backward compatibility with older Megatron-Bridge versions

### PR DESCRIPTION
**Pull request by nuerxiati**
### What does this PR do?

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

After #6335 (`bc3f3bf0`), verl adopted several new Megatron-Bridge APIs that do not exist in older versions. Using the latest verl with an older Megatron-Bridge causes `ImportError` and `AttributeError` at runtime.

- Add backward-compatible fallbacks in three files so that the latest verl works with both new and old Megatron-Bridge versions without requiring a Bridge upgrade.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [x] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: verl/models/mcore/bridge.py [hunk 1]
```diff
@@ -15,12 +15,91 @@
 
 try:
```
Comment (Mind-s) [inline verl/models/mcore/bridge.py:16]: releases/v0.8.0是否也可以增加该版本兼容PR

Comment (nuerxiati) [reply verl/models/mcore/bridge.py:16]: 已提：https://github.com/verl-project/verl/pull/6695

File: verl/models/mcore/bridge.py [hunk 2]
```diff
@@ -15,12 +15,85 @@
...
     from megatron.bridge import AutoBridge
-    from megatron.bridge.training.utils.train_utils import LinearForLastLayer, freeze_moe_router, make_value_model
 except ImportError:
-    # `pip install verl[mcore]` or
     print("Megatron-Bridge package not found. Please install Megatron-Bridge with `pip install megatron-bridge`")
     raise
 
+try:
+    from megatron.bridge.training.utils.train_utils import LinearForLastLayer, freeze_moe_router, make_value_model
```
Comment (wucong25) [inline verl/models/mcore/bridge.py:25]: 加上注释表明分支兼容情况

File: verl/models/mcore/bridge.py [hunk 3]
```diff
@@ -15,12 +15,85 @@
 
 try:
     from megatron.bridge import AutoBridge
```
Comment (wucong25) [inline verl/models/mcore/bridge.py:17]: 原pr中from megatron.bridge.models.conversion.param_mapping import AutoMapping，AutoMapping.register_module_type("LinearForLastLayer", "replicated")这段代码确认一下是否不需要

File: verl/models/mcore/bridge.py [hunk 4]
```diff
@@ -15,12 +15,85 @@
...
+                            and layer.mlp.shared_experts.gate_weight is not None
+                        ):
+                            layer.mlp.shared_experts.gate_weight.requires_grad = False
+                        if (
+                            hasattr(layer.mlp.shared_experts, "gate_bias")
+                            and layer.mlp.shared_experts.gate_bias is not None
+                        ):
+                            layer.mlp.shared_experts.gate_bias.requires_grad = False
+        return model
+
```
Comment (wucong25) [inline verl/models/mcore/bridge.py:102]: lora相关代码均已删除，不支持lora兼容

File: verl/utils/megatron_utils.py [hunk 5]
```diff
@@ -248,19 +248,8 @@ def make_megatron_module(
...
-            from megatron.bridge.training.utils.config_utils import create_ddp_config
-
-            # When using PEFT with Megatron-Bridge, we must apply PEFT transformation
-            # BEFORE wrapping the model in DDP. This is required because:
-            # 1. PEFT freezes base model parameters (requires_grad=False)
-            # 2. DDP must be aware of which parameters are trainable when building gradient buckets
-            # 3. The distributed optimizer must only track trainable (adapter) parameters
-            # See Megatron-Bridge docs: training/peft.md
-
-            # Register PEFT transformation as pre-wrap hook if peft_cls is specified
```
Comment (wucong25) [inline verl/utils/megatron_utils.py:349]: 这段注释为啥删除了，尽量不要影响gpu相关逻辑和注释

File: verl/utils/megatron_utils.py [hunk 6]
```diff
@@ -292,12 +281,33 @@ def peft_info_hook(model):
...
-            ddp_config = create_ddp_config(
-                wrap_with_ddp=wrap_config.wrap_with_ddp,
-                use_distributed_optimizer=wrap_config.use_distributed_optimizer,
-                use_megatron_fsdp=wrap_config.use_megatron_fsdp,
-                overrides=override_ddp_config,
-            )
+            ddp_config = None
+            if wrap_config.wrap_with_ddp:
+                try:
+                    from megatron.bridge.training.utils.config_utils import create_ddp_config
```
Comment (wucong25) [inline verl/utils/megatron_utils.py]: 这里也加上注释

File: verl/utils/megatron_utils.py (hunk 5 above)
Comment (nuerxiati) [reply verl/utils/megatron_utils.py:349]: 已恢复

File: verl/utils/megatron_utils.py (hunk 6 above)
Comment (nuerxiati) [reply verl/utils/megatron_utils.py]: done

File: verl/models/mcore/bridge.py (hunk 2 above)
Comment (nuerxiati) [reply verl/models/mcore/bridge.py:25]: done

**Review by wucong25 — DISMISSED** [review]

File: verl/models/mcore/bridge.py [hunk 7]
```diff
@@ -15,12 +15,93 @@
...
 try:
     from megatron.bridge import AutoBridge
-    from megatron.bridge.training.utils.train_utils import LinearForLastLayer, freeze_moe_router, make_value_model
 except ImportError:
-    # `pip install verl[mcore]` or
     print("Megatron-Bridge package not found. Please install Megatron-Bridge with `pip install megatron-bridge`")
     raise
 
+# Megatron-Bridge >= v0.5.0 exposes these symbols in train_utils.
+# Megatron-Bridge <  v0.5.0 does not, so we fall back to local implementations.
```
Comment (ji-huazhong) [inline verl/models/mcore/bridge.py:23]: Why is compatibility with the legacy version of megatron-bridge still required?

File: verl/models/mcore/bridge.py [hunk 8]
```diff
@@ -15,12 +15,92 @@
...
```
Comment (wuxibin89) [inline verl/models/mcore/bridge.py:23]: Which megatron-bridge version mindspeed currently use?

**Review by wuxibin89 — APPROVED** [review]
